### PR TITLE
Continuing development of ZIPLN models

### PR DESCRIPTION
**Pull request by jchiquet**
- [x] Exact optimization with Lambert function for $R$ (parameter of the conditional distirbution of the ZI component)
- [x] Network-like methods for ZIPLNfit_sparse
- [x] ZIPLNnetwork, a 
    - [x] collection with PLNfamilly collecting a series of ZIPLNfit_sparse object with automatic grid
    - [x] merge/share classes between PLNnetworkfamily and ZIPLNnetwork

### Still missing

- [ ] use correct weights in estimation of $\pi$ when using exact distributino of $W$ (with Lambert function)
- [x] complete test for ziplnnetwork

Note that I won't add postTreatment yet zo ZIPLN since I think we need to rethink a bit all the psotTreatment made on PLN-lie objects(in particular, all the variance estimators...)

### CODE REVIEW

**Review by mahendra-mariadassou — COMMENTED** [review]
Apart from the 3 small comments:
- inception_cov to constrain the shape of the covariance matrix in the inception model
- the remark about lapply / future.lapply
- number of parameters for ZIPLNnetwork (the extra p is indeed present in PLNnetwork)
It's all good for me.

File: R/ZIPLNfit-class.R [hunk 1]
```diff
@@ -730,31 +716,102 @@ ZIPLNfit_fixed <- R6Class(
...
+    #' @field penalty the global level of sparsity in the current model
+    penalty         = function() {private$lambda},
+    #' @field penalty_weights a matrix of weights controlling the amount of penalty element-wise.
+    penalty_weights = function() {private$rho},
+    #' @field n_edges number of edges if the network (non null coefficient of the sparse precision matrix)
+    n_edges         = function() {sum(private$Omega[upper.tri(private$Omega, diag = FALSE)] != 0)},
     #' @field nb_param number of parameters in the current PLN model
     nb_param   = function() {
-      res <-  self$p * self$d + (sum(private$Omega != 0) - self$p)/2L +
+      res <-  self$p * self$d + self$n_edges +
```
[the remark this answers is not in the document]
Comment (mahendra-mariadassou) [reply R/ZIPLNfit-class.R]: Now applies to
https://github.com/PLN-team/PLNmodels/blob/36123fd4c8229ef8ae25b258a30a37670502dcd6/R/ZIPLNfit-class.R#L804-L806

File: R/PLNnetwork.R [hunk 2]
```diff
@@ -122,5 +122,6 @@ PLNnetwork_param <- function(
     variance          = TRUE             ,
     config_post       = config_pst       ,
     config_optim      = config_opt       ,
+### TODO CHECK: Why two inceptive model (cov and not ?)
```
[the remark this answers is not in the document]
Comment (jchiquet) [reply R/PLNnetwork.R:126]: noted, thx.

File: R/PLNnetworkfamily-class.R [hunk 3]
```diff
@@ -366,3 +271,270 @@ PLNnetworkfamily <- R6Class(
...
+      if (is.null(subsamples)) {
+        subsample.size <- round(ifelse(private$n >= 144, 10*sqrt(private$n), 0.8*private$n))
+        subsamples <- replicate(20, sample.int(private$n, subsample.size), simplify = FALSE)
+      }
+
+      ## got for stability selection
+      cat("\nStability Selection for PLNnetwork: ")
+      cat("\nsubsampling: ")
+
+      stabs_out <- future.apply::future_lapply(subsamples, function(subsample) {
```
[the remark this answers is not in the document]
Comment (jchiquet) [reply R/PLNnetworkfamily-class.R:369]: I used `lapply` in place of `future_lapply` for debugging (the later suppress the output which is not very convenient for debuging). For the moment, I suggest we use future_lapply for all parallel computing, and we will update this particular instance once we will have our complete rethinking of all parallel computing stuff.

**Review by mahendra-mariadassou — APPROVED** [review]
All good for me (and I should indeed have looked at it sooner...)

**Pull request merged**